flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/SchemaOperator.java

```diff
@@ -609,6 +643,34 @@ public Object getFieldOrNull(RecordData recordData) {
                                                     + "Currently only CHAR / VARCHAR can be accepted by a STRING column",
                                             originalField.getClass())));
                 }
+            } else if (destinationType.is(DataTypeRoot.TIME_WITHOUT_TIME_ZONE)
+                    && originalType.is(DataTypeRoot.TIME_WITHOUT_TIME_ZONE)) {
+                Integer value = (Integer) originalField;
+                int precisionDifference =
+                        ((TimeType) originalType).getPrecision()
+                                - ((TimeType) destinationType).getPrecision();
+                if (precisionDifference > 0) {
+                    return value / (int) Math.pow(10, precisionDifference);
+                } else if (precisionDifference < 0) {
+                    return value * (int) Math.pow(10, -precisionDifference);
+                } else {
+                    return value;
+                }
```

Comment:
Currently, time supports only a precision of 0. We need to open another PR to implement a new feature that   custom precision for time, and then implement a new feature that support merging time columns when routing.

Reply:
Agree that `TIME` precision is a different issue and could be dealt with in another PR.